ddtrace/internal/processor/trace.py

```diff
@@ -127,6 +130,10 @@ def process_trace(self, trace):
 
         ctx._update_tags(chunk_root)
         chunk_root.set_tag_str("language", "python")
+        # for 128 bit trace ids
+        if chunk_root.trace_id > _MAX_UINT_64BITS:
```

Comment:
this check bothers me a bit because it relies on the implementation of the "random" 128 bit trace id is generated. Since it uses the unix time stamp (a non-zero value) we're guaranteed that the generated 128bit id is > _MAX_UINT_64BITS. If implementation were to change and we truly generated random 128 bit trace ids then this check would not hold and this code would be broken.

I'd like to see a stronger variant that we use to establish that a span is using 128-bit ids. This could be done if we upfront stored the 128 trace id as a tag in the span on span creation and keep `span.trace_id` as just the 64 bit portion. Then here we would check the presence of that tag on the span.

WDYT @mabdinur?

Reply:
note that this also might be more performant as it would be less costly to access the 64 bit trace id. We also wouldn't have to change all the places that we reference `trace_id` in our code. Also also, this would not be a breaking change for users who are assuming that trace ids are 64 bits.

We could then introduce a backwards compatible `span.trace_id_128` to get the 128 bit id

Reply:
Even if we generate a traceid of say 10 when 128bit trace if generation is enabled we don’t need to set or extract from the tag containing higher order bits. We only need to set and extract _dd.p.tid if a trace id is greater than 2**64 - 1. 

Trace_id is referenced in many places in our library. If we introduce two interfaces for setting and accessing a trace ids (ex: traceid and traceid128) it would significantly increase the complexity of using Spans. The current approach was designed with this in mind. Also the higher order bits tags should not be set on every span. It should only be set on the chunk root span. This was defined in the RFC

Reply:
TLDR this code won’t break even if we generate a 64bit trace id when DD_TRACE_128_Traceid_generation=true.

Reply:
To clarify, by "broken" I mean that we won't set the HOB tag in traces and then any assumptions around that would break eg. if the backend relies on that tag to determine if 128 bit trace ids are being generated. This is a weak example though. I just want to make sure that this is a safe condition to use.

👍 makes sense that the logic holds when _using_ the trace id.

Reply:
The RFC states that the `_dd.p.tid` tag should only be set if there are higher order bits. So this is inline with the spec. I think it's safe condition to use.

Reply:
:+1: 👍